### ISSUE TITLE
[Flutter GPU] Use unsigned Dart FFI types where appropriate.

### ISSUE THREAD

**Issue by bdero** [flutter/flutter#157129]
Dart FFI supports unsigned types like Uint32. So we should use those instead of int types when passing things like enums to the engine.

**Comment by jonahwilliams** [comment]
I don't think you really need unsigned types if the values are already "sanitized". Like if the enum int comes from the dart package passing value.index, you know that is safe to cast to the enum.

But if you ask for an int and only positive or non-zero values are legal - then there needs to be a runtime check somewhere

**Comment by bdero** [comment]
These values are mostly sanitized or otherwise guaranteed positive on the Dart side. My thought is it would be nice to get rid of all the questionable-looking casts on the cpp side.
